views/api.go

```diff
@@ -29,35 +30,35 @@ type (
 )
 
 // SetTokenHandler sets a valid JWT in a cookie instead of returning a string
-func (v *Views) SetTokenHandler(w http.ResponseWriter, r *http.Request) {
-	//fmt.Println("DEBUG - SET TOKEN")
-	session, _ := v.cookie.Get(r, v.conf.SessionCookieName)
+func (v *Views) SetTokenHandler(c echo.Context) error {
+	session, _ := v.cookie.Get(c.Request(), v.conf.SessionCookieName)
 	u := helpers.GetUser(session)
 
 	tokenString, err := v.newJWT(u)
 	if err != nil {
 		err = fmt.Errorf("failed to set cookie: %w", err)
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
+		http.Error(c.Response(), err.Error(), http.StatusInternalServerError)
+		return err
```

Comment:
`return fmt.Errorf(...)` and let the error handler handle the rest. Similar for the other places where we do this.

Reply:
Will do that in a bit

Reply:
Sorted